tests/phpunit/CRM/MembersOnlyEvent/Hook/Copy/EventFromTemplateCreatorTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use CRM_MembersOnlyEvent_Test_Fabricator_Event as EventFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_MembersOnlyEvent as MembersOnlyEventFabricator;
+use CRM_MembersOnlyEvent_BAO_MembersOnlyEvent as MembersOnlyEvent;
+
+require_once __DIR__ . '/../../../../BaseHeadlessTest.php';
+
+/**
+ * Class CRM_MembersOnlyEvent_BAO_MembersOnlyEventTest
+ *
+ * @group headless
+ */
+class CRM_MembersOnlyEvent_Hook_Copy_EventFromTemplateCreatorTest extends BaseHeadlessTest {
+
+
+  /**
+   * Tests Create MemberOnlyEvent when new Event was created from template.
+   */
+  public function testEventFromTemplate() {
```

Comment:
I would rename this to : 

testCreatingEventFromTemplate();

and would probably add another test method to test that setAllowMembershipTypesIds do its job.

Reply:
Another test method should not be done as it is a private method.

Reply:
We can test it by adding the membership type and assert if the membeship types have been added correctly.

Reply:
I did not mean to test the private method but the public create method, 

> We can test it by adding the membership type and assert if the membeship types have been added correctly.

Yes but I would create another test method for that, to do the same as the first one (you can move the common code into private method) and then add the membership types then assert. like that you separate your test cases better.